### PR TITLE
Move nav_goals_generator from strands_util

### PR DESCRIPTION
**Pull request by ferdianjovan**
This is needed to fix https://github.com/strands-project/strands_hri/pull/79

### CODE REVIEW

**Comment by cdondrup** [comment]
I suggested to put this here, as it seems to fit quite well. If there are objections we can still move it to strands_apps, but I don't really see why atm.
Maybe one of the main maintainers, @BFALacerda or @Jailander, can make a qualified statement. Or @kunzel as the author.

**Comment by Jailander** [comment]
I think it could go here too, on other comments is a pity @gestom and @santosj didn't know this existed since they have a component that does the same.

**Comment by cdondrup** [comment]
Has been sitting in `strands_utils` for a while ;)

File: nav_goals_generator/launch/nav_goals_generator.launch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<launch>
+    <arg name="map" default="/map"/>
+    <arg name="is_costmap" default="false"/>
+    <arg name="machine" default="localhost" />
+    <arg name="user" default="" />
+
+    <machine name="$(arg machine)" address="$(arg machine)" env-loader="/opt/strands/strands_catkin_ws/devel/env.sh" user="$(arg user)" default="true"/>
```
Comment (cdondrup) [inline nav_goals_generator/launch/nav_goals_generator.launch]: @ferdianjovan, just to save @Jailander the work of refactoring it later, could you please change this line to something similar to [this](https://github.com/Jailander/strands_navigation/blob/hydro-devel/joy_map_saver/launch/joy_map_saver.launch#L12) before we merge this?

**Comment by Jailander** [comment]
Just saying

File: nav_goals_generator/launch/nav_goals_generator.launch (hunk 1 above)
Comment (Jailander) [inline nav_goals_generator/launch/nav_goals_generator.launch]: yes but with the machine tag like this please:

```
<machine name="$(arg machine)" address="$(arg machine)" env-loader="(optenv ROS_ENV_LOADER )" user="$(arg user)" default="true"/>
```

**Comment by ferdianjovan** [comment]
Is that ok now?

**Comment by cdondrup** [comment]
Not sure about the new line... @jailander?

**Comment by marc-hanheide** [comment]
what new line?

**Comment by ferdianjovan** [comment]
Sorry about that, vim does it automatically. Did not see it before.

**Comment by marc-hanheide** [comment]
@cdondrup @Jailander do you want to re-release this once it passed the tests?

**Comment by bfalacerda** [comment]
i also have a couple of additions to make, part of getting the recovery behaviours ready for release. Can you wait till later today?

**Comment by Jailander** [comment]
yes I can wait until all pull requests are merged, @hawesie also openned one right now

**Comment by cdondrup** [comment]
@ferdianjovan can you please remove the `nav_goals_generator` from `strands_utils` now that it's here? Thanks.

Yes, please release when everything is merged so I can see if `strands_hri` builds.

**Comment by ferdianjovan** [comment]
Actually the nav_goals_msgs is inside nav_goals_generator now. Should I also remove nav_goals_msgs from strands_utils?

**Pull request merged**